### PR TITLE
Add base69 decoder

### PR DESCRIPTION
**Pull request by lukasgabriel**
As seen here: https://github.com/pshihn/base69
My code is, apart from the Ciphey-specific stuff, a 1:1 translation of the JavaScript original.
---
There is one major problem that needs to be fixed before the decoder (hopefully) works as intended:
For some reason, the output values in positions 2-7 are offset by 256.

Here's an example: 
I encode the string `AAAAAAA` with [the base69 demo.](https://pshihn.github.io/base69/)
This produces the encoded string `gALBoAUAKAFACA*A`
When I run my base69 decoder on this string, its output (in decimal form) is the following:
`[65, 321, 321, 321, 321, 321, 321]`. 
As 65 is the decimal representation of an `'A'`, the expected output would be:
 `[65, 65, 65, 65, 65, 65, 65]`.

65 + 256 = 321 - I figured out that much on myself :) This relationship holds for other input strings as well. 
For some reason, the first character out of 7 is always correct.

Being a total novice in all matters text encoding, bitwise operations and all that stuff I am at a loss here 😄 so I'll happily accept any help here. Honestly I don't really understand what the << and >> operators do at all.

When I was trying to fix it myself, I put in a few print statements so it's easier to trace the steps the decoder takes and where it trips up (and learn about the bitwise operators in the process).

### CODE REVIEW

**Comment by SkeletalDemise** [comment]
Nice. I was also trying to implement a Base69 decoder. I think I got stuck with the same bug as you did. Things weren't working right even though I did a close to 1:1 Python implementation of the JavaScript so I kinda gave up. 
Also, here's some information about bitwise operators that might be useful:
https://wiki.python.org/moin/BitwiseOperators
https://www.w3schools.com/js/js_bitwise.asp

**Comment by lukasgabriel** [comment]
@SkeletalDemise thanks a lot! I'm sure we can find a solution if we look into it more. We can always just ask the dev of the original for advice.

File: ciphey/basemods/Decoders/base69.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
+# community
```
Comment (brandonskerritt) [inline ciphey/basemods/Decoders/base69.py:1]: You are no longer community so don't include this + your github repo :) 

This was purely so we could track down who wrote the original code easily && for our tag system that's coming soon:tm:

File: ciphey/basemods/Decoders/base69.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        return (69 * self.CHARS.index(s[1])) + (self.CHARS.index(s[0]))
+
+    def decode_chunk(self, s: str):
+        padded_bytes = s.endswith('=')
+        decoded = [0 for _ in range(8)]
+        for i in range(8):
+            decoded[i] = 0 if i == 7 and padded_bytes else self.chars_to_byte(s[ i * 2 : i * 2 + 2 ])
+            print(f'decoded[{i}] = {decoded[i]}')
+        result = [0 for _ in range(7)]
+        for i in range(7):
```
Comment (brandonskerritt) [inline ciphey/basemods/Decoders/base69.py:29]: this looks hard O.O

**Review by brandonskerritt — CHANGES_REQUESTED** [review]

**Comment by Cyclic3** [comment]
The code you are translating from uses a byte array, so all operations happen mod 256. You can easily (and hackily!) replicate this behaviour by just running `x % 256` on each output byte `x` when you've decoded it.

**Comment by Cyclic3** [comment]
The README for base69 says that you can work out how many null bytes were added using the penultimate character. This would mean that we could decode data such as 0x01000100  without losing the terminal null byte

**Comment by lukasgabriel** [comment]
> 
> 
> The README for base69 says that you can work out how many null bytes were added using the penultimate character. This would mean that we could decode data such as 0x01000100 without losing the terminal null byte

@Cyclic3 can you tell me what I'd need to change to make that work? 😄

**Comment by Cyclic3** [comment]
I think you just need to read the n-1'th character, parse it as an integer, and trim that many bytes off it

**Comment by lukasgabriel** [comment]
Finally working & thoroughly tested. Should work without issues with the vast majority of inputs. Collision with other decoders (namely reverse) will be solved when the new checker arrives.

**Review by bee-san — APPROVED** [review]

**Pull request merged**